src/main/java/com/vint/shop/controllers/AdminController.java

```diff
@@ -0,0 +1,38 @@
+package com.vint.shop.controllers;
+
+import com.vint.shop.service.impl.UserDetailsServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class AdminController {
+    @Autowired
+    private UserDetailsServiceImpl userDetailsServiceImpl;
+
+    @GetMapping("/admin")
+    public String userList(Model model) {
+        model.addAttribute("allUsers", userDetailsServiceImpl.allUsers());
+        return "admin";
+    }
+
+    @PostMapping("/admin")
+    public String  deleteUser(@RequestParam(required = true, defaultValue = "" ) Long userId,
+                              @RequestParam(required = true, defaultValue = "" ) String action,
+                              Model model) {
+        if (action.equals("delete")){
+            userDetailsServiceImpl.deleteUser(userId);
+        }
+        return "redirect:/admin";
+    }
+
+    @GetMapping("/admin/gt/{userId}")
```

Comment:
Something strange.
It's not REST application, it's MVC application and you should use request params instead of PathVariable

Reply:
this is a sketch